### PR TITLE
Certora comments (and reorder functions)

### PR DESCRIPTION
**Pull request by EndymionJkb**

### CODE REVIEW

File: pkg/pool-weighted/contracts/smart/InvestmentPool.sol [hunk 1]
```diff
@@ -115,11 +115,11 @@ contract InvestmentPool is BaseWeightedPool, ReentrancyGuard {
 
     // External functions
 
-    /**
-     * @dev Indicates whether swaps are enabled or not for the given pool.
+    /*
+     * @dev Can enable/disable trading
```
Comment (TomAFrench) [inline pkg/pool-weighted/contracts/smart/InvestmentPool.sol:119]:
```suggestion
     * @dev Set whether swaps are enabled or not for the given pool.
```

To match the style of other comments

File: pkg/pool-weighted/contracts/smart/InvestmentPool.sol [hunk 2]
```diff
@@ -158,14 +158,6 @@ contract InvestmentPool is BaseWeightedPool, ReentrancyGuard {
...
-        return _MAX_WEIGHTED_TOKENS;
-    }
-
-    function _getTotalTokens() internal view virtual override returns (uint256) {
-        return _getMiscData().decodeUint7(_TOTAL_TOKENS_OFFSET);
-    }
-
     /**
      * @dev Schedule a gradual weight change, from the current weights to the given endWeights,
      * over startTime to endTime
```
Comment (TomAFrench) [inline pkg/pool-weighted/contracts/smart/InvestmentPool.sol:163]:
```suggestion
     * over startTime to endTime.
```

File: pkg/pool-weighted/contracts/smart/InvestmentPool.sol [hunk 3]
```diff
@@ -399,12 +397,6 @@ contract InvestmentPool is BaseWeightedPool, ReentrancyGuard {
...
     }
 
-    function _readScalingFactor(bytes32 tokenState) private pure returns (uint256) {
-        uint256 decimalsDifference = tokenState.decodeUint5(_DECIMAL_DIFF_OFFSET);
-
-        return FixedPoint.ONE * 10**decimalsDifference;
-    }
-
     /**
      * @dev Extend ownerOnly functions to include the Investment Pool control functions
```
Comment (TomAFrench) [inline pkg/pool-weighted/contracts/smart/InvestmentPool.sol:401]:
```suggestion
     * @dev Extend ownerOnly functions to include the Investment Pool control functions.
```

**Review by TomAFrench — APPROVED** [review]
Looks good to me. I've made a couple of suggestions on comments to standardise them a little.

File: pkg/pool-weighted/contracts/smart/InvestmentPool.sol [hunk 4]
```diff
@@ -115,11 +115,11 @@ contract InvestmentPool is BaseWeightedPool, ReentrancyGuard {
 
     // External functions
 
-    /**
-     * @dev Indicates whether swaps are enabled or not for the given pool.
+    /*
```
Comment (TomAFrench) [inline pkg/pool-weighted/contracts/smart/InvestmentPool.sol:118]:
```suggestion
    /**
```
Nitpicky as hell.

Comment (EndymionJkb) [reply pkg/pool-weighted/contracts/smart/InvestmentPool.sol:118]: That's actually semantically different (NatSpec or not). Good catch!

**Comment by nventuro** [comment]
Superceded by https://github.com/balancer-labs/balancer-v2-monorepo/pull/822

**Pull request closed without merging**